src/WebJobs.Script/Extensions/HttpRequestExtensions.cs

```diff
@@ -89,12 +91,24 @@ public static bool IsColdStart(this HttpRequest request)
 
         public static Uri GetRequestUri(this HttpRequest request) => new Uri(request.GetDisplayUrl());
 
-        public static byte[] GetRequestBodyAsBytes(this HttpRequest request)
+        public static async Task<byte[]> GetRequestBodyAsBytes(this HttpRequest request)
         {
-            var length = Convert.ToInt32(request.ContentLength);
-            var bytes = new byte[length];
-            request.Body.Read(bytes, 0, length);
```

Comment:
For OOP, there were a lot of places where we were reading the request synchronously. I've made all of this async, and it resulted in huge perf gains